infinispan-remote/src/test/java/org/hibernate/ogm/datastore/infinispanremote/utils/InfinispanRemoteTestHelper.java

```diff
@@ -68,6 +69,30 @@ public long getNumberOfAssociations(SessionFactory sessionFactory) {
 		return totalCount;
 	}
 
+	@Override
+	public long getNumberOfAssociations(Session session) {
+		final InfinispanRemoteDatastoreProvider datastoreProvider = getProvider( session.getSessionFactory() );
+		final SessionFactoryImplementor sessionFactoryImplementor = getSessionFactoryImplementor( session.getSessionFactory() );
+		Collection<CollectionPersister> persisters = sessionFactoryImplementor.getMetamodel().collectionPersisters().values();
+		long totalCount = 0;
+		for ( CollectionPersister ep : persisters ) {
+			OgmCollectionPersister persister = (OgmCollectionPersister) ep;
+			String tableName = persister.getTableName();
+			SingleTableOgmEntityPersister owningPersister = (SingleTableOgmEntityPersister) persister.getOwnerEntityPersister().getEntityPersister();
+			String ownerTableName = owningPersister.getTableName();
+
+			RemoteCache<Object, Object> cache = datastoreProvider.getCache( tableName );
+			if ( !( cache instanceof TransactionalRemoteCacheImpl ) ) {
+				totalCount += countAssociations( tableName, ownerTableName, datastoreProvider );
+				continue;
```

Comment:
A small note, could you use a if-then-else instead of the `continue`, please? I find the code easier to follow without the `continue`.

Reply:
Sure